### PR TITLE
Flatpak for Bustle

### PR DESCRIPTION
**Pull request by wjt**
This is taken directly from https://cgit.freedesktop.org/bustle/tree/flatpak?h=bustle-0.6

### CODE REVIEW

File: org.freedesktop.Bustle.json [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        },
+
+        /* fired up! ready to go! */
+        {
+            "name": "bustle",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://anongit.freedesktop.org/git/bustle.git",
+                    "branch": "bustle-0.6"
```
Comment (TingPing) [inline org.freedesktop.Bustle.json]: Needs to include a specific commit or specific tarball.

File: org.freedesktop.Bustle.json [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        "/bin/pcap-config"
+    ],
+    "modules": [
+        /* Why not start your day with a 259MB tarball? */
+        {
+            "name": "haskell-platform",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://www.haskell.org/platform/download/8.0.2/haskell-platform-8.0.2-unknown-posix--full-x86_64.tar.gz",
```
Comment (alexlarsson) [inline org.freedesktop.Bustle.json]: This is x86-64 only, so you need to create a flathub.json with skip-arches for the other arches.
Alternatively, you could have only-arch on this module and add version for the other arches.
Here is an example flathub.json: https://github.com/flathub/org.DolphinEmu.dolphin-emu/blob/master/flathub.json

Comment (wjt) [inline org.freedesktop.Bustle.json]: Okay. The Haskell Platform bundle appears to only be available for i386 and x86_64 so I guess I will need `{ "skip-arches": ["arm", "aarch64"]}` in any case.

**Comment by wjt** [comment]
Updated to build on i386, skip aarch64 and arm, and point explicitly to 0.6.1 by tag and commit hash.

**Comment by Barthalion** [comment]
Test build started [here](https://flathub.org/builds/#/builders/1/builds/170).

**Comment by Barthalion** [comment]
Looks like our buildbot is unhappy about C-style comments (although I think it wasn't before). Poked @ramcq on IRC about it.

**Comment by wjt** [comment]
Well, on the one hand, it's totally legit to complain about them: they are not valid JSON, and @ebassi tells me that json-glib only allows them by accident.

On the other hand, almost every manifest I've seen uses them. Maybe Flathub insisting that JSON means JSON will change that.

**Comment by ebassi** [comment]
> Maybe Flathub insisting that JSON means JSON will change that.

It's *highly* unlikely.

Maybe we should use another format for user-writable manifests, and then "compile" it to JSON.

**Comment by wjt** [comment]
Here's a technique I've used in other places where I have to write JSON by hand but want to add a comment. Valid JSON is (modulo edge cases, please don't tell me about weird whitespace or surrogate pairs) valid YAML, and YAML supports `# ` comments. So I just renamed the manifest, replaced `/* */` with `#`, and wrote a Python program to convert it back.

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "skip-arches": ["arm", "aarch64"]
```
Comment (ramcq) [inline flathub.json]: We've now added only-arches to this file syntax. \o/

File: update-manifest.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python3
+import yaml
+import json
+
+
+def main():
+    '''The master copy is only YAML so we can have comments.'''
```
Comment (ramcq) [inline update-manifest.py]: We should now accept
/* multi-line
 * c-style
 * comments */

**Comment by wjt** [comment]
@ramcq fixed, if you want to test the multi-line comment fix

**Comment by Barthalion** [comment]
The [build](https://flathub.org/builds/#/builders/1/builds/175) succeeded – I'll merge it later today.

**Comment by Barthalion** [comment]
And with small delay, the repo has been created. Thanks!

**Pull request closed without merging**